Assets/Scripts/Map/MapPatrollingGen/WatchmanRouteSolver.cs

```diff
@@ -102,7 +103,7 @@ private static List<Vector2Int> ComputeVertexCoordinates(Dictionary<Vector2Int,
             return guardPositions;
         }
 
-        private static Dictionary<Vector2Int, HashSet<Vector2Int>> ComputeVisibility(bool[,] map)
+        private static Dictionary<Vector2Int, HashSet<Vector2Int>> ComputeVisibility(bool[,] map, System.Func<Vector2Int, bool[,], HashSet<Vector2Int>> visibilityAlgorithm)
```

Comment:
It should probably be defined as a delegate instead of specifying all type parameters everywhere it is used.